### PR TITLE
Add printf attributes to CmiPrintf, CmiError, CmiAbort

### PR DESCRIPTION
**Pull request by evan-charmworks**
Please rebase this PR as a merge strategy

### CODE REVIEW

File: src/libs/ck-libs/idxl/idxl.C [hunk 1]
```diff
@@ -38,7 +30,7 @@ CLINKAGE void pupIDXL_Chunk(pup_er cp) {
 }
 IDXL_Chunk *IDXL_Chunk::get(const char *callingRoutine) {
 	IDXL_Chunk *c=getNULL();
-	if(!c) IDXL_Abort(callingRoutine,"IDXL is not initialized");
+	if(!c) IDXL_Abort("IDXL is not initialized", callingRoutine);
```
Comment (matthiasdiener) [inline src/libs/ck-libs/idxl/idxl.C]: Since IDXL_Abort was removed, should this be CkAbort?

Comment (matthiasdiener) [reply src/libs/ck-libs/idxl/idxl.C]: nvm, saw the defined below

**Comment by matthiasdiener** [comment]
Fails on Darwin:

```
./converse.h:1800:1: error: an attribute list cannot appear here
CMK_NORETURN void CmiAbort(const char *msg, ...);
^~~~~~~~~~~~
./converse.h:135:24: note: expanded from macro 'CMK_NORETURN'
#  define CMK_NORETURN [[noreturn]]
                       ^~~~~~~~~~~~
1 error generated.
```

**Comment by evan-charmworks** [comment]
```
$ git diff bed69ba6b..10f21daf4
diff --git a/src/conv-core/converse.h b/src/conv-core/converse.h
index beb2dd13b..a4b615f0d 100644
--- a/src/conv-core/converse.h
+++ b/src/conv-core/converse.h
@@ -1794,10 +1794,13 @@ void realConverseExit(int exitcode);
 #if CMK_SHRINK_EXPAND
 void ConverseCleanup(void);
 #endif
+
+CMK_NORETURN
 #if defined __GNUC__ || defined __clang__
 __attribute__ ((format (printf, 1, 2)))
 #endif
-CMK_NORETURN void CmiAbort(const char *msg, ...);
+void CmiAbort(const char *msg, ...);
+
 void CmiOutOfMemory(int nBytes);
 
 #if CMK_MEMCHECK_OFF
```

File: benchmarks/converse/pingpong/pingpong.C [hunk 2]
```diff
@@ -43,12 +43,12 @@ void startRing()
...
 {
+  size_t msgSizeDiff = CpvAccess(msgSize)-CmiMsgHeaderSizeBytes;
   CmiFree(msg);
 
   //Print the time for that message size
-  CmiPrintf("Size=%d bytes, time=%lf microseconds one-way\n",
-      CpvAccess(msgSize)-CmiMsgHeaderSizeBytes,
-      (1e6*(CpvAccess(endTime)-CpvAccess(startTime)))/(2.*CpvAccess(nCycles)));
+  CmiPrintf("Size=%zu bytes, time=%lf microseconds one-way\n",
+      msgSizeDiff, (1e6*msgSizeDiff)/(2.*CpvAccess(nCycles)));
```
Comment (rbuch) [inline benchmarks/converse/pingpong/pingpong.C]: The value for time looks wrong here.

Comment (evan-charmworks) [reply benchmarks/converse/pingpong/pingpong.C]: Good catch, thanks! Fixed.

**Pull request merged**